### PR TITLE
Start both virtlogd and libvirtd foreground

### PR DESCRIPTION
**Pull request by yafu-1**
When starting libvirtd foreground, it needs to start virtlogd foregroud
too. Also modify the function run_kill_virsh_while_managedsave() to
exit the gdb correctly.

Signed-off-by: Yan Fu <yafu@redhat.com>

### CODE REVIEW

**Comment by yafu-1** [comment]
Test results:
```
# avocado run --vt-type libvirt daemon.crash_regression
JOB ID     : a37bed9d5716db85c4a25ae32c87d6c5ff7a1106
JOB LOG    : /root/avocado/job-results/job-2017-07-26T09.57-a37bed9/job.log
TESTS      : 11
 (01/11) type_specific.io-github-autotest-libvirt.daemon.crash_regression.destroy_console: PASS (11.92 s)
 (02/11) type_specific.io-github-autotest-libvirt.daemon.crash_regression.shutdown_console: PASS (85.23 s)
 (03/11) type_specific.io-github-autotest-libvirt.daemon.crash_regression.restart_save_restore: PASS (39.05 s)
 (04/11) type_specific.io-github-autotest-libvirt.daemon.crash_regression.mix_boot_order_os_boot: PASS (11.09 s)
 (05/11) type_specific.io-github-autotest-libvirt.daemon.crash_regression.kill_virsh_while_managedsave: PASS (179.39 s)
 (06/11) type_specific.io-github-autotest-libvirt.daemon.crash_regression.job_acquire: PASS (26.94 s)
 (07/11) type_specific.io-github-autotest-libvirt.daemon.crash_regression.invalid_interface: PASS (11.61 s)
 (08/11) type_specific.io-github-autotest-libvirt.daemon.crash_regression.cpu_compare: PASS (10.66 s)
 (09/11) type_specific.io-github-autotest-libvirt.daemon.crash_regression.pm_test: PASS (91.73 s)
 (10/11) type_specific.io-github-autotest-libvirt.daemon.crash_regression.invalid_mac_net: PASS (10.76 s)
 (11/11) type_specific.io-github-autotest-libvirt.daemon.crash_regression.restart_firewalld: PASS (14.91 s)
RESULTS    : PASS 11 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0
JOB HTML   : /root/avocado/job-results/job-2017-07-26T09.57-a37bed9/html/results.html
TESTS TIME : 493.30 s

```

File: libvirt/tests/src/daemon/crash_regression.py [hunk 1]
```diff
@@ -101,17 +105,19 @@ def run_kill_virsh_while_managedsave(params, libvirtd, vm):
...
+    #    target=virsh.managedsave,
+    #    args=(vm.name,)
+    #)
+    #thread.start()
+    #try:
+    #   res = utils.run('pkill -9 virsh', ignore_status=True)
+    #   if res.exit_status:
+    #       raise error.TestError("Failed to kill virsh:\n%s", res)
+    #finally:
+    #    thread.join()
```
Comment (balamuruhans) [inline libvirt/tests/src/daemon/crash_regression.py]: please remove the comments that is not necessary

Comment (yafu-1) [inline libvirt/tests/src/daemon/crash_regression.py]: Thanks for reminding.  I have removed the comments. Please help to review again.

File: libvirt/tests/src/daemon/crash_regression.py [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 from virttest import virsh
 from virttest import data_dir
 from virttest.utils_libvirtd import LibvirtdSession
 from virttest.libvirt_xml.xcepts import LibvirtXMLError
 from virttest.libvirt_xml.vm_xml import VMXML, VMCPUXML, VMPMXML
 from virttest.libvirt_xml.network_xml import NetworkXML, IPXML
 from virttest.libvirt_xml.devices import interface
+
 from autotest.client.shared import error
 from autotest.client import utils
```
Comment (balamuruhans) [inline libvirt/tests/src/daemon/crash_regression.py]: can you help in changing `utils` with `process` that is imported. and `error` can be removed

File: libvirt/tests/src/daemon/crash_regression.py [hunk 3]
```diff
@@ -247,7 +242,13 @@ def run(test, params, env):
     vm_name = params.get("main_vm", "virt-tests-vm1")
     bug_url = params.get("bug_url", None)
     vm = env.get_vm(vm_name)
-
+    #Run virtlogd foreground
+    try:
+        path.find_command('virtlogd')
+        process.run("systemctl stop virtlogd", ignore_status=True)
```
Comment (balamuruhans) [inline libvirt/tests/src/daemon/crash_regression.py]: do we have to start virtlogd in foreground , but instead we are stopping it.

Comment (yafu-1) [inline libvirt/tests/src/daemon/crash_regression.py]: Yes,but we need to stop the virtlogd started by systemctl firstly, then start virtlogd foreground.

File: libvirt/tests/src/daemon/crash_regression.py [hunk 4]
```diff
@@ -101,17 +105,8 @@ def run_kill_virsh_while_managedsave(params, libvirtd, vm):
...
-    )
-    thread.start()
-    try:
-        res = utils.run('pkill -9 virsh', ignore_status=True)
-        if res.exit_status:
-            raise error.TestError("Failed to kill virsh:\n%s", res)
-    finally:
-        thread.join()
+    process.run("virsh managedsave %s &" % (vm.name), shell=True)
+    process.run("pkill -9 virsh", ignore_status=True)
```
Comment (balamuruhans) [inline libvirt/tests/src/daemon/crash_regression.py]: let us check the status and raise on error.

File: libvirt/tests/src/daemon/crash_regression.py (hunk 3 above)
Comment (balamuruhans) [inline libvirt/tests/src/daemon/crash_regression.py]: Okay, thanks for clarifying.

File: libvirt/tests/src/daemon/crash_regression.py (hunk 4 above)
Comment (yafu-1) [inline libvirt/tests/src/daemon/crash_regression.py]: Both 'virsh managedsave vm.name &‘ and 'pkill -9 virsh' may be failed to execute, such as:
1.'pkill -9 virsh' kill managedsave successfully, 'virsh managedsave vm.name &' will be failed.
2.If 'pkill -9 virsh' executed after virsh managedsave vm.name completed(sometimes virsh managedsave complete very quick), then 'pkill -9 virsh' will be failed.
So, IMO, it's no need to check the status, the test will be repeated 100 times, which will include all kinds of scenarios.

**Comment by yafu-1** [comment]
@balamuruhans , I have updated the code. Could you help to review it again please? Thanks.

File: libvirt/tests/src/daemon/crash_regression.py (hunk 4 above)
Comment (balamuruhans) [inline libvirt/tests/src/daemon/crash_regression.py]: okay.

**Comment by balamuruhans** [comment]
Thanks @yafu-1

**Pull request merged**